tools.sh

```diff
@@ -128,5 +134,13 @@ function start_vm {
     done
 }
 
+function generate_htpasswd_file {
+   local auth_file_dir=$1
+   local pass_file=$2
+   random_password=$(cat $1/auth/kubeadmin-password)
```

Comment:
Can you call this `kubeadmin_password` or such? `random_password` when we don't generate it was confusing to me.